src/dependencies/InstallerSequence.ts

```diff
@@ -13,18 +13,34 @@ export class InstallerSequence {
 		}, []);
 	}
 
-	public async install(location: Location, shouldUpdate: boolean, progressListener: ProgressListener): Promise<Location> {
+	public async install(location: Location, shouldUpdate: boolean, progressListener: ProgressListener, confirm:() => Promise<ConfirmResult>): Promise<InstallResult<Location>> {
 		let index = 0;
+		let askedForConfirmation = false;
+		let result: InstallResult<Location> = new Success(location);
 		const total = this.installers.length;
 		for (const installer of this.installers) {
-			location = await installer.install(location, shouldUpdate, (fraction, message) => {
+			function intermediateListener(fraction: number, message: string) {
 				progressListener(
 					(index + fraction) / total,
 					`${message} (step ${index + 1} of ${total})`
 				);
-			});
+			}
+			function intermediateConfirm(): Promise<ConfirmResult> {
+				// only ask once
+				if (askedForConfirmation) {
+					return Promise.resolve(ConfirmResult.Continue);
```

Comment:
I think this should resolve to the old result of the `confirm()` call, otherwise only the first installer will actually cancel the execution.

Reply:
All subsequent installers will be skipped as soon as a `cancel` has been observed

Reply:
Actually, I think that it's the *promise* of the old `confirm()` call that should be returned. In fact, the returned promise shouldn't resolve before the user makes a choice.

Reply:
Mmm, I see your point. However, that needs a new requirement on `DependencyInstaller`, doesn't it? Like, each installer should call or pass to another step `cancel` at most once. The documentation should say whether it's fine to call `cancel` multiple times or not.

Reply:
(Also, the two `intermediate*` functions could be moved outside the loop, but it doesn't change the semantics.)